Client/Frontend/Browser/NightModeHelper.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import Shared
+
+struct NightModePrefsKey {
+    static let NightModeButtonIsInMenu = PrefsKeys.KeyNightModeButtonIsInMenu
+    static let NightModeStatus = PrefsKeys.KeyNightModeStatus
+}
+
+class NightModeHelper: TabHelper {
+
+    private weak var tab: Tab?
+
+    required init(tab: Tab) {
+        self.tab = tab
+        if let path = NSBundle.mainBundle().pathForResource("NightModeHelper", ofType: "js"), source = try? NSString(contentsOfFile: path, encoding: NSUTF8StringEncoding) as String {
+            let userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.AtDocumentStart, forMainFrameOnly: true)
+            tab.webView!.configuration.userContentController.addUserScript(userScript)
+        }
+    }
+
+    static func name() -> String {
+        return "NightMode"
+    }
+
+    func scriptMessageHandlerName() -> String? {
+        return "NightMode"
+    }
+
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        // Do nothing.
+    }
+
+    static func isNightModeAvailable(state: AppState) -> Bool {
```

Comment:
Move this method and the one after to a `NightModeAccessors` class in the same file. 

I should've done this for `NoImageMode` but I didn't spot it.